benchmark/src/main/java/com/google/api/gax/grpc/CallableBenchmark.java

```diff
@@ -0,0 +1,126 @@
+import io.grpc.Status.Code;
+import java.io.ByteArrayOutputStream;
+import java.util.Collections;
+import java.util.List;
+import org.joda.time.Duration;
+
+public class CallableBenchmark {
+  private static final TopicName TOPIC_NAME_RESOURCE = TopicName.create("my-project", "my-topic");
+  private static final String TOPIC_NAME_STRING = TOPIC_NAME_RESOURCE.toString();
+  private static final FutureCallable<PublishRequest, Integer> RETURN_ONE_CALLABLE =
+      new FutureCallable<PublishRequest, Integer>() {
+        @Override
+        public RpcFuture<Integer> futureCall(PublishRequest request, CallContext context) {
+          return new ListenableFutureDelegate<Integer>(Futures.immediateFuture(1));
+        }
+      };
+  private static final RetrySettings RETRY_SETTINGS =
+      RetrySettings.newBuilder()
+          .setTotalTimeout(Duration.standardSeconds(1))
+          .setInitialRetryDelay(Duration.standardSeconds(1))
+          .setRetryDelayMultiplier(1.2)
+          .setMaxRetryDelay(Duration.standardSeconds(1))
+          .setInitialRpcTimeout(Duration.standardSeconds(1))
+          .setRpcTimeoutMultiplier(1.2)
+          .setMaxRpcTimeout(Duration.standardSeconds(1))
+          .build();
+  private static final UnaryCallable<PublishRequest, Integer> ONE_UNARY_CALLABLE =
+      UnaryCallable.create(RETURN_ONE_CALLABLE)
+          .retryableOn(ImmutableSet.of(Code.UNAVAILABLE))
+          .retrying(RETRY_SETTINGS, null);
+  private static final List<PubsubMessage> MESSAGES = createMessages();
+
+  private static List<PubsubMessage> createMessages() {
+    ImmutableList.Builder<PubsubMessage> messages = ImmutableList.builder();
+    for (int i = 0; i < 100; i++) {
+      messages.add(
+          PubsubMessage.newBuilder()
+              .setData(ByteString.copyFromUtf8(String.format("test-message-%d", i)))
+              .build());
+    }
+    return messages.build();
+  }
+
+  private int serialize(int reps, int numMessages) throws Exception {
+    int totalSize = 0;
+    for (int i = 0; i < reps; i++) {
+      PublishRequest request =
+          PublishRequest.newBuilder()
+              .setTopic(TOPIC_NAME_STRING)
+              .addAllMessages(MESSAGES.subList(0, numMessages))
+              .build();
+      ByteArrayOutputStream out = new ByteArrayOutputStream();
+      request.writeTo(out);
+      totalSize += out.size();
+    }
+    return totalSize;
+  }
+
+  @Benchmark
+  int timeSerialize0(int reps) throws Exception {
+    return serialize(reps, 0);
+  }
+
+  @Benchmark
+  int timeSerialize10(int reps) throws Exception {
+    return serialize(reps, 10);
+  }
+
+  @Benchmark
+  int timeSerialize100(int reps) throws Exception {
+    return serialize(reps, 100);
+  }
+
+  @Benchmark
+  int timeRequest(int reps) {
+    int total = 0;
+    for (int i = 0; i < reps; i++) {
+      total +=
+          ONE_UNARY_CALLABLE.call(
+              PublishRequest.newBuilder().setTopicWithTopicName(TOPIC_NAME_RESOURCE).build());
+    }
```

Comment:
I feel like the scope of these tests is more limited than what is useful to get the full picture - is this just supposed to be a baseline for the components to isolate where problems are if they show up? Where is the test going to be for a call through a generated GAPIC client? (I don't think it goes in GAX)